README.md

```diff
@@ -44,6 +44,7 @@ The Wasm-sdk's build process needs some packages :
 * `clang`
 * `ninja`
 * `python3`
+* The latest rust toolchain
```

Comment:
Looks like [this](https://github.com/bytecodealliance/wasm-component-ld/pull/36/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542R9) would indicate that 1.76.0 should work? But you mention that 1.77.0 did not work for you... what did you see?

Reply:
see https://github.com/WebAssembly/wasi-sdk/issues/450

Reply:
@abrown do you have any more concerns?

Reply:
Well, I kind of was thinking we could be more precise here and since `wasm-component-ld` is being successfully with Rust 1.76.0 in CI we should probably document that. But then I'm confused why 1.77.0 didn't work for you? Or was something else the problem?

Reply:
i dunno. i merely blindly followed alex's advice to update the toolchain.

but is it worth to block this PR for a month while we don't document such precise verisions for other requirements either? (cmake python etc)

Reply:
I don't think it's worth merging an inaccurate claim such as this because it's not true that "the latest rust toolchain" is required to build wasm-component-ld. I think it's better to either rename this to `rustc`, as the other tools are here, or work towards a stricter requirement here such as [updating the project to test a minimum version](https://github.com/bytecodealliance/wasm-component-ld/pull/36) and then listing that version here.

Reply:
I sure wasn't trying to block any PRs, it's just that something here didn't make sense so I moved on to other things. Now that I've circled back, I think since it isn't clear to you which version you used but CI would indicate that the minimum supported version is 1.76.0, we can document that. That way we can avoid telling users to do extra toolchain updates; it sounds like the extra overhead of new toolchain updates is something you wouldn't be happy with.

Reply:
[oops, commented on a stale page and didn't see @alexcrichton's reply; I suggested an update that should resolve this]

Reply:
> I don't think it's worth merging an inaccurate claim such as this because it's not true that "the latest rust toolchain" is required to build wasm-component-ld.

Sorry, which part of the claim was incorrect?
"Latest"?
1.77.0 didn't work and you adviced me I was using too old toolchian. So I updated it to 1.79.0 and it worked.
Isn't it inappropriate to call 1.79.0 latest?

Reply:
> since it isn't clear to you which version you used

It's clear it was 1.77.0.

Reply:
Ok, I guess I was the one who misunderstood, sorry! But something still doesn't seem quite right with what you reported. I tried to replicate what you found and I'm seeing that `wasm-component-ld` builds just fine with version v1.76.0:

```console
$ cargo install 1.76.0
$ cargo default 1.76.0
$ cargo version
cargo 1.76.0 (c84b36747 2024-01-18)
$ cmake -G Ninja -B build/toolchain -S . -DWASI_SDK_BUILD_TOOLCHAIN=ON -DCMAKE_INSTALL_PREFIX=build/install
...
$ cmake --build build/toolchain --target wasm-component-ld
...
   Compiling wasm-component-ld v0.5.6
    Finished release [optimized] target(s) in 18.50s
  Installing /.../wasi-sdk/build/toolchain/wasm-component-ld/bin/wasm-component-ld
   Installed package `wasm-component-ld v0.5.6` (executable `wasm-component-ld`)
```

If I run the same with v1.75.0, I see the following failure:

```
error: cannot install package `wasm-component-ld 0.5.6`, it requires rustc 1.76.0 or newer, while the currently active rustc version is 1.75.0
```

Is there any chance something else was wrong with the environment you ran v1.77.0 in?

Reply:
> Sorry, which part of the claim was incorrect? "Latest"?

Yes, this is incorrect. For example when you created this PR Rust 1.79 was the latest Rust release. I believe what you were trying to document originally was that Rust 1.79 was required. As of today, however, Rust 1.80 is the latest Rust release. That means that the claim that the "latest" Rust release is needed is not correct.

Additionally `wasm-component-ld` as-used by wasi-sdk requires 1.76.0, and as Andrew pointed out this is tested in CI. This additionally means that the claim that the latest compiler is required is incorrect.

Reply:
> Ok, I guess I was the one who misunderstood, sorry! But something still doesn't seem quite right with what you reported. I tried to replicate what you found and I'm seeing that `wasm-component-ld` builds just fine with version v1.76.0:
> 
> ```
> $ cargo install 1.76.0
> $ cargo default 1.76.0
> $ cargo version
> cargo 1.76.0 (c84b36747 2024-01-18)
> $ cmake -G Ninja -B build/toolchain -S . -DWASI_SDK_BUILD_TOOLCHAIN=ON -DCMAKE_INSTALL_PREFIX=build/install
> ...
> $ cmake --build build/toolchain --target wasm-component-ld
> ...
>    Compiling wasm-component-ld v0.5.6
>     Finished release [optimized] target(s) in 18.50s
>   Installing /.../wasi-sdk/build/toolchain/wasm-component-ld/bin/wasm-component-ld
>    Installed package `wasm-component-ld v0.5.6` (executable `wasm-component-ld`)
> ```
> 
> If I run the same with v1.75.0, I see the following failure:
> 
> ```
> error: cannot install package `wasm-component-ld 0.5.6`, it requires rustc 1.76.0 or newer, while the currently active rustc version is 1.75.0
> ```
> 
> Is there any chance something else was wrong with the environment you ran v1.77.0 in?

i can still reproduce https://github.com/WebAssembly/wasi-sdk/issues/450 with rustc 1.77.0 + wasm-component-ld 0.5.5.
(we bumped wasm-component-ld version to 0.5.6, which doesn't seem to have the problem, after this PR.)

Reply:
> Yes, this is incorrect. For example when you created this PR Rust 1.79 was the latest Rust release. I believe what you were trying to document originally was that Rust 1.79 was required. As of today, however, Rust 1.80 is the latest Rust release. That means that the claim that the "latest" Rust release is needed is not correct.

ok. i dropped the "latest" claim. the main claim of this PR was "we require rust".